cypress/integration/api-user-creation.spec.js

```diff
@@ -0,0 +1,119 @@
+           auth: {
+                username: Cypress.env('WP_USERNAME'),
+                password: Cypress.env('WP_PASSWORD')
+           },
+           body:users.user_with_no_username,
+           failOnStatusCode: false
+        }).should((response) => {
+            expect(response.status).to.eq(400)
+            expect(response.body).to.have.property('code', 'rest_missing_callback_param')
+            expect(response.body).to.have.property('message', 'Missing parameter(s): username')
+        })          
+    })
+
+    it('should not create a user with no password', function() {
+        cy.request({
+           method: 'POST', 
+           url: endpoints.user, 
+           auth: {
+                username: Cypress.env('WP_USERNAME'),
+                password: Cypress.env('WP_PASSWORD')
+           },
+           body:users.user_with_no_password,
+           failOnStatusCode: false
+        }).should((response) => {
+            expect(response.status).to.eq(400)
+            expect(response.body).to.have.property('code', 'rest_missing_callback_param')
+            expect(response.body).to.have.property('message', 'Missing parameter(s): password')
+        })          
+    })
+
+    it('should not create a user with no email', function() {
+        cy.request({
+           method: 'POST', 
+           url: endpoints.user, 
+           auth: {
+                username: Cypress.env('WP_USERNAME'),
+                password: Cypress.env('WP_PASSWORD')
+           },
+           body:users.user_with_no_email,
+           failOnStatusCode: false
+        }).should((response) => {
+            expect(response.status).to.eq(400)
+            expect(response.body).to.have.property('code', 'rest_missing_callback_param')
+            expect(response.body).to.have.property('message', 'Missing parameter(s): email')
+        })          
+    })
+
+    it('should not create a user with no authentication credentials', function() {
+        cy.request({
+           method: 'POST', 
+           url: endpoints.user, 
+           body:users.user,    
+           failOnStatusCode: false
+        }).should((response) => {
+            expect(response.status).to.eq(401)
+            expect(response.body).to.have.property('message', 'Sorry, you are not allowed to create new users.')
+            expect(response.body).to.have.property('code', 'rest_cannot_create_user')
+        })          
+    })
+
+    it('should delete an existing user', function(){
+        cy.request({     
+            method: 'DELETE', 
+            url: `${endpoints.user}${this.id}?reassign=false&force=true`, 
+            auth: {
+                username: Cypress.env('WP_USERNAME'),
+                password: Cypress.env('WP_PASSWORD')
+            }
+        }).should((response) => {
+            expect(response.status).to.eq(200)
+            expect(response.body).to.have.property('deleted', true)
+            expect(response.body.previous).to.have.property('id', this.id)
```

Comment:
You're using a variable you defined on a previous test (`this.id`), but this creates a dependency between tests that makes them flaky, because if the test that initializes this `id` variable fails before setting it up, then this particular test will fail as well